src/tinygraph.c

```diff
@@ -31,10 +31,14 @@ TINY_GRAPH *TinyGraphConnect(TINY_GRAPH *G, int i, int j)
 {
     if(TinyGraphAreConnected(G, i, j))
 	return G;
+    if(i==j) assert(G->selfLoops);
     TSetAdd(G->A[i], j);
-    if(!G->directed) TSetAdd(G->A[j],i);
+
     ++G->degree[i];
-    if(i==j) assert(G->selfLoops);
+    if(!G->directed){
+	TSetAdd(G->A[j],i);
+	++G->degree[j];
```

Comment:
This will incorrectly increment degree TWICE if selfLoops is true. Please re-arrange logic to avoid this.